axelor-human-resource/src/main/java/com/axelor/apps/hr/service/dpae/DPAEServiceImpl.java

```diff
@@ -0,0 +1,410 @@
+  }
+
+  public void sendSingle(DPAE dpae) throws AxelorException {
+    if (exportSingle(dpae) == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(IExceptionMessage.DPAE_ALREADY_SEND),
+          dpae);
+    }
+    Message message = sendMessageDPAE(dpae.getMetaFile(), dpae);
+    sentDPAEService.createSentDPAE(message, dpae.getMetaFile(), Arrays.asList(dpae.getEmployee()));
+  }
+
+  protected String exportSingle(DPAE dpae) throws AxelorException {
+    if (dpae.getIsSent()) {
+      return null;
+    }
+    List<String> listErrors = checkDPAEValidity(dpae);
+    if (!listErrors.isEmpty()) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_MISSING_FIELD,
+          String.format(I18n.get(IExceptionMessage.DPAE_FIELD_INVALID), listErrors.toString()),
+          dpae);
+    }
+    String textString = generateTextString(dpae);
+
+    try {
+      AppSettings appSettings = AppSettings.get();
+      String exportDir = appSettings.get("data.export.dir");
+      if (exportDir == null) {
+        throw new IllegalArgumentException(I18n.get("Export directory is not configured."));
+      }
+      File file = new File(exportDir + "/" + dpae.getRegistrationDPAE() + ".txt");
+      try (PrintWriter writer = new PrintWriter(file.getAbsoluteFile(), "UTF-8")) {
+        writer.print(textString);
+      }
+
+      // Attach Meta File to DPAE
+      try (InputStream is = new FileInputStream(file)) {
+        DMSFile dmsFile = Beans.get(MetaFiles.class).attach(is, file.getName(), dpae);
+        dpae.setMetaFile(dmsFile.getMetaFile());
+      }
+      dpae.setIsSent(true);
+      return textString;
+    } catch (Exception e) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR, e.getMessage(), dpae);
+    }
+  }
+
+  protected Message sendMessageDPAE(MetaFile fileToSend, DPAE dpae) throws AxelorException {
+    try {
+      Template templateDPAE =
+          Beans.get(AppHumanResourceService.class).getAppEmployee().getTemplateDPAE();
+      if (templateDPAE == null) {
+        throw new AxelorException(
+            DPAE.class,
+            TraceBackRepository.CATEGORY_MISSING_FIELD,
+            IExceptionMessage.DPAE_MISSING_TEMPLATE);
+      }
+      templateDPAE = Beans.get(TemplateRepository.class).find(templateDPAE.getId());
+      Message message =
+          Beans.get(TemplateMessageService.class)
+              .generateMessage(
+                  dpae.getId(),
+                  DPAE.class.getCanonicalName(),
+                  DPAE.class.getSimpleName(),
+                  templateDPAE);
+      messageService.attachMetaFiles(message, ImmutableSet.of(fileToSend));
+      message = Beans.get(MessageService.class).sendByEmail(message);
+      return message;
+    } catch (Exception e) {
+      throw new AxelorException(
+          DPAE.class, TraceBackRepository.CATEGORY_CONFIGURATION_ERROR, e.getMessage());
```

Comment:
I think if an exception occur for example in generate message, the saved traceback will not contain the real error and instead will reference this line, making debug harder. is this possible to remove this try catch clause and instead throw any errors ?